### ISSUE TITLE
Redundant trait bounds

### ISSUE THREAD

**Issue by regexident** [Manishearth/rust-clippy#873]
From time to time I find redundancies in my code as the result of moving trait bounds around while refactoring.

While true redundancy detection (with indirection) is probably out of scope …

```rust
trait Baz {}
trait Bar: Baz {}

trait Foo<T>
	where T: Bar {
	fn foo() where T: Baz;
}
```

… trivial cases like this should be totally doable:

```rust
trait Bar {}

trait Foo<T>
	where T: Bar {
	fn foo() where T: Bar;
}
```

**Comment by llogiq** [comment]
Good idea.

As long as we have the information that `Bar` is somehow bound by `Baz`, I don't even see a problem with linting your first example. However, I'm unsure if this can work if the traits in question are in another crate.

**Comment by regexident** [comment]
You're probably right. I keep forgetting that linters are more than just AST pattern matchers. 😇
